### PR TITLE
Mobile: Improve syntax highlighting

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
Adds syntax highlighting for code blocks, block quotes, math in the mobile beta editor.
Screenshot of a math note in an emulator

This is a refactored version of #6607.

# Additional screenshots
Screenshot of the sample text, light mode enabled
Screenshot of the sample text, dark mode enabled

# Testing
The text used in the screenshot is the following:
<details><summary>Text used in <code>additional screenshots</code> section</summary>

````markdown
# Summary
 * The goal of this note is to demonstrate $\TeX$, `code`, and _other_ styling in the beta editor.

# $\TeX$ Highlighting
As can be seen, $\text{both numbers (e.g. 0, 1, 2)}$ and commands are highlighted. The highlighter uses [lezer](https://lezer.codemirror.net) to do parsing (it's the backend that CodeMirror 6 uses).

The math highlighting supports both inline and block math. For example,
> The Cauchy integral theorem states that, given some $z \in \mathbb{C}$ and $r \in \mathbb{R}_{\geq 0}$,
> if $f : \mathbb{C} \to \mathbb{C}$ is analytic in $B_r(z_0)$, then,
> $$
>   f(z) = \oint_{\partial B_r(z_0)} \frac{f(w)}{(w - z)^{n+1}} dw
> $$

Testing escaped: $\$$. It works!
$$
	% As a result of syntax highlighting, comments are also styled.
	\text{\TeX works inside and outside of block quotes.}
$$

<!-- More block math: -->
$$ 3 + 3 \neq 7 $$

$3+3 = 6$

# `Code` highlighting
```html
<h1>Code highlighting</h1>
<p>This is an example of highlighted HTML.</p>

$$
  \TeX isn't highlighted in HTML blocks.
$$
```

```
Code with no language.
```

```js
"use strict";
(function() {
	alert("Testing!");
})();
```
````

</details>

### CODE REVIEW

File: packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	public update(viewUpdate: ViewUpdate) {
+		if (viewUpdate.docChanged || viewUpdate.viewportChanged) {
+			this.decorations = computeDecorations(viewUpdate.view);
+		}
+	}
+}, {
+	decorations: pluginVal => pluginVal.decorations,
+});
+
+export default decoratorPlugin;
```
Comment (laurent22) [inline packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts]: Shouldn't naming be consistent here? The file is called "decorator", it exports a "decoratorPlugin" variable, which is then imported as "decoratorExtension". Which one of these is the most accurate?

File: packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+});
+
+const headerLineDecoration = Decoration.line({
+	attributes: { class: 'cm-headerLine' },
+});
+
+/**
+ * @return a set of [Decoration]s, associated with block syntax groups that require
+ * full-line styling.
+ */
```
Comment (laurent22) [inline packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts]: Please use // comments

File: packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
+/**
+ * Exports an editor plugin that creates multi-line decorations based on the
+ * editor's syntax tree (assumes markdown).
+ *
+ * For more about creating decorations, see https://codemirror.net/examples/zebra/
+ */
```
Comment (laurent22) [inline packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts]: // comments

File: packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts (hunk 1 above)
Comment (personalizedrefrigerator) [reply packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts]: Good point! The file should be named `decoratorExtension` and export `decoratorExtension`.

File: packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts (hunk 2 above)
Comment (personalizedrefrigerator) [reply packages/app-mobile/components/NoteEditor/CodeMirror/decorators.ts]: Changed! Thank you for the reminder!

**Comment by laurent22** [comment]
The quality of the mobile editor is really going up with your recent changes, thanks for adding this!

**Pull request merged**